change to ssh and limit snap use

Immaannn2222 opened the pull request:
changed from https to ssh for git cloning
use pipx instead of pip

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,66 @@
+
+# Display public key
+cat ~/.ssh/id_ed25519.pub
+echo "Copy the above key to your GitHub"
```
where in my Github?

Immaannn2222 replied:
Ok I'll be more explicit ans state it is in https://github.com/settings/keys and done one time only

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,66 @@
+# Clone repositories
+git clone git@github.com:khaldoun-xyz/lugha.git
+cd lugha
+pipx install --include-deps --pip-args="-r requirements.txt" .
```
with `pipx` I don't need dedicated virtualenvs anymore? How would I quickly switch between my virtualenv for sisu to my virtualenv in lugha, if they don't exist anymore? Or are the python packages bound to the directory I'm in??

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,66 @@
 sudo apt update -y
 sudo apt upgrade -y
 
-sudo apt-get install snapd -y                                     # snap store
```
where do all of these installs go?

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,66 @@
-deactivate
-
-# neovim
-curl -LO https://github.com/neovim/neovim/releases/latest/download/nvim-linux-x86_64.appimage
```
where do all of these installs go?

s0288 reviewed (commented):
I think you deleted some of my installs :(. Or am I misunderstanding sth?

Immaannn2222 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,66 @@
-mkdir ~/programming/
+# Install pipx
+sudo apt install pipx -y
+pipx ensurepath
```
I couldn't install anything with pip and pipx was always recommended to solve the problem of multiple virtual envorionments.
I, myself never used before also

Immaannn2222 commented:
Hello for the newest commit:

- Added the rest of the tools as the original setup 
- used pipx where it should 
- added logging methods and helper functions
- kept the SSh method but enhanced 
- some minor conflicts remain but I was able to fully install it

s0288 reviewed (commented):
I think this is a great structural improvement to the status quo. 
However, I am not able to run the install script yet (Alacritty install fails). 

After we merge this, this file will be moved to Altadim's repo, correct?
Will you then also update Altadim's `README.md` to describe how to run the install script as a user?

s0288 commented:
for completeness' sake, I'm also publishing my error with Alacritty here: 
![image](https://github.com/user-attachments/assets/1d739106-3ea0-4703-abf4-ee6bba9e9ccd)

Immaannn2222 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,406 @@
+        setup_python_project_ssh "${khaldoun_ssh_repos[$project]}" "$project"
+    done
+
+    # Special case for core_skills (now also using SSH URL)
```
It does not need a  virtual environment and `pip install -r requirements.txt`

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,406 @@
+
+# Check if the script is run with root privileges
+if [[ $EUID -ne 0 ]]; then
+    error_exit "This script must be run with sudo or as root. Please run 'sudo ./khaldoun-setup.sh'"
```
I meant that in the last line of this section we currently specify `bash khaldoun-setup.sh`.

s0288 commented on setup/khaldoun-setup.sh:
```diff
@@ -1,100 +1,406 @@
+        log "         If you plan to clone Khaldoun projects via SSH, you need to generate an SSH key"
+        log "         and add the public key to your GitHub account."
+        log "         You can generate a new key with: ssh-keygen -t ed25519 -C \"your_email@example.com\""
+        log "         See GitHub's documentation for adding the key to your account."
```
I noticed now that we already discussed this 3 weeks ago (!!). Please, let's not keep having the same conversations several times. Once is enough.

> Ok I'll be more explicit ans state it is in https://github.com/settings/keys and done one time only

s0288 replied:
@Immaannn2222

s0288 reviewed (approved):
Merging this now to make progress and move it to the altadim repo.
